python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -281,6 +343,188 @@ def launch_scheduled_runs(
         yield error_info
 
 
+def _process_tick(
+    workspace_process_context: IWorkspaceProcessContext,
+    logger: logging.Logger,
+    external_schedule: ExternalSchedule,
+    schedule_state: InstigatorState,
+    schedule_state_lock: threading.Lock,
+    end_datetime_utc: datetime.datetime,
+    max_catchup_runs: int,
+    max_tick_retries: int,
+    tick_retention_settings,
+    schedule_debug_crash_flags,
+    log_verbose_checks,
+):
+    # evaluate the tick immediately, but from within a thread.  The main thread should be able to
+    # heartbeat to keep the daemon alive
+    list(
+        _process_tick_generator(
+            workspace_process_context,
+            logger,
+            external_schedule,
+            schedule_state,
+            schedule_state_lock,
+            end_datetime_utc,
+            max_catchup_runs,
+            max_tick_retries,
+            tick_retention_settings,
+            schedule_debug_crash_flags,
+            log_verbose_checks,
+        )
+    )
+
+
+def _process_tick_generator(
```

Comment:
i think these names are worse than the previous ones - the code in question doesn't line up with the sensors behavior quite the same